components/ws-daemon/cmd/run.go

```diff
@@ -94,6 +108,21 @@ var runCmd = &cobra.Command{
 			go pprof.Serve(cfg.PProf.Addr)
 		}
 
+		if cfg.ReadinessProbeAddr != "" {
+			// Ensure we can resolve DNS queries, can access gcr.io and the local Registry is up and running
+			health := healthcheck.NewHandler()
+			wsManagerFQDN := fmt.Sprintf("ws-manager.%v.svc.cluster.local", os.Getenv("KUBE_NAMESPACE"))
+			health.AddReadinessCheck("dns", kubernetes.DNSCanResolveProbe(wsManagerFQDN, 1*time.Second))
+			health.AddReadinessCheck("grpc-server", grpcProbe(cfg.Service))
+			health.AddReadinessCheck("ws-daemon", dmn.ReadinessProbe())
```

Comment:
What's the motivation for this check? ws-daemon does not connect to ws-manager

Reply:
removed